syncstorage/tests/functional/test_storage.py

```diff
@@ -1575,7 +1575,7 @@ def test_batch_with_failing_bsos(self):
         endpoint = collection + '?batch={0}&commit=true'.format(batch)
         resp = self.app.post_json(endpoint, bsos)
         self.assertEqual(len(resp.json['failed']), 1)
-        self.assertEqual(len(resp.json['success']), 1)
+        self.assertEqual(len(resp.json['success']), 2)
```

Comment:
I think what "success" means needs some clarity. I would expect to interpret the results like this: 

First Request: 

- `a` was successfully appended to the batch
- `b` failed and reasons are listed in `failed`. In this case, with a bad id

The response tells me what was successful/failed to be appended to a batch.

Second Request:

- `a` was successfully committed
- `c` was successfully committed
- `d` failed due to reasons and was not committed 

With a `commit=true` the response tells me what was successfully committed to the database. The data is now officially in the collections.

The ambiguity would make it harder to implement clients correctly. If Request #1 sends a success and then the batch times out and is purged for some reason what does that mean to the client? 

If there's already client code that is "done", I'm willing to acquiesce to that behavior if I have to. :)